### PR TITLE
Improve outline buttons

### PR DESCRIPTION
**Pull request by maya**
This updates outline buttons to use an inset box shadow for it's outline, rather than using a border and having to reduce the padding. This ensures the buttons remain the same size no matter the size of the outline someone chooses. 

It's less good for old IE but since we are reducing support, I felt this was a good approach bc of it's increased flexibility and ease of use.

Also this PR fixes the homepage mobile button bug in #576 and https://github.com/18F/web-design-standards/issues/463#issuecomment-140472125 (but not that issue, they are separate problems).

### CODE REVIEW

File: assets/_scss/elements/_buttons.scss [hunk 1]
```diff
@@ -103,40 +103,36 @@ button,
...
-    border: 2px solid;
-    border-color: $color-primary;
+    box-shadow: inset 0 0 0 2px $color-primary;
     color: $color-primary;
-    padding: .8rem 1.8rem;
 
     &:hover,
     &.usa-button-hover {
-      border-color: $color-primary-darker;
+      box-shadow: inset 0 0 0 2px $color-primary-darker;
```
Comment (msecret) [inline assets/_scss/elements/_buttons.scss]: Is there a way to store `inset 0 0 0 2px` in a variable to use in all the places rather the repeating it?

Comment (maya) [inline assets/_scss/elements/_buttons.scss]: @msecret good point, do you think we should store in variables.scss or the top of the buttons file?

Comment (msecret) [inline assets/_scss/elements/_buttons.scss]: Top of buttons file

Comment (maya) [inline assets/_scss/elements/_buttons.scss]: @msecret done in c88892c.

**Pull request merged**